caddyhttp/markdown/markdown.go

```diff
@@ -97,94 +101,81 @@ func (md Markdown) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error
 		return http.StatusMethodNotAllowed, nil
 	}
 
+	// TODO: Deal with dirents variable in new implementation
 	var dirents []os.FileInfo
-	var lastModTime time.Time
-	fpath := r.URL.Path
-	if idx, ok := httpserver.IndexFile(md.FileSys, fpath, cfg.IndexFiles); ok {
-		// We're serving a directory index file, which may be a markdown
```

Comment:
Where has all this logic gone?  Is it no longer needed?

Reply:
I believe this functionality is handled by staticfiles.

Reply:
While staticfiles seems to handle the standard situation of index.html or other index files that have been configured outside of the markdown directive, the existing markdown code does have a config of IndexFiles that includes index.md, index.markdown, and index.mdown by default and would have something like index.ext extensions were set in the markdown configuration.

Is there a way to send this list of possible index files up the change when calling this?
https://github.com/jimjimovich/caddy/blob/b0f4085af3f02d289144c6a84e739109cdf00ae3/caddyhttp/markdown/markdown.go#L135-L136

Reply:
Perhaps having index files defined in the markdown plugin is actually a bad idea and they should instead be defined in http.index?

https://caddyserver.com/docs/index

Reply:
Yes, definitely let the staticfiles package figure out index files. :+1: